### PR TITLE
Include JSON representation of upcoming and past conferences

### PR DESCRIPTION
**Pull request by ashdavies**
Whilst playing with a pet project that would list conferences, I found it would be useful to have this information in JSON format, as the alternative required querying the GitHub API.

Sorry for not accompanying this with a feature request or making an issue first, the implementation turned out to be fairly trivial.

~~I couldn't figure out how to preserve a line break after a closing bracket though.~~

### CODE REVIEW

**Review by cortinico — APPROVED** [review]
LGTM but want to hear from @cketti as well before we merge this

File: past.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    "website": "{{ conference.website }}",
+    "location": "{{ conference.location }}",
+    {%- if conference.status %}
+    "status": "{{ conference.status }}",
+    {%- endif -%}
+    {%- if conference.online %}
+    "online": true,
+    {%- endif %}
+    "dateStart": "{{ conference.date_start | date: "%Y-%m-%d" }}",
+    "dateEnd": "{{ conference.date_end | date: "%Y-%m-%d" }}",
```
Comment (cketti) [inline past.json]: This will lead to a trailing comma (that's not allowed in JSON) when there's no CfP information.

File: past.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    {%- if conference.cfp.start %}
+    "cfp": {
+      "start": "{{ conference.cfp.start | date: "%Y-%m-%d" }}",
+      "end": "{{ conference.cfp.end | date: "%Y-%m-%d" }}",
+      "site": "{% if conference.cfp.site %}{{ conference.cfp.site }}{% else %}{{ conference.website }}{% endif %}"
+    }
+    {%- endif -%}
```
Comment (cketti) [inline past.json]: See https://shopify.github.io/liquid/basics/whitespace/

```suggestion
    {%- endif %}
```

File: past.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    }{% unless forloop.last %}, {% endunless %}
```
Comment (cketti) [inline past.json]:
```suggestion
  }{% unless forloop.last %}, {% endunless %}
```

File: past.json [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  {%- endif -%}
+{%- endfor -%}
```
Comment (cketti) [inline past.json]:
```suggestion
{%- endfor %}
```

File: past.json [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    "location": "{{ conference.location }}",
+    {%- if conference.status %}
+    "status": "{{ conference.status }}",
+    {%- endif -%}
+    {%- if conference.online %}
+    "online": true,
+    {%- endif %}
+    "dateStart": "{{ conference.date_start | date: "%Y-%m-%d" }}",
+    "dateEnd": "{{ conference.date_end | date: "%Y-%m-%d" }}",
+    {%- if conference.cfp.start %}
```
Comment (cketti) [inline past.json]:
```suggestion
    "dateEnd": "{{ conference.date_end | date: "%Y-%m-%d" }}"
    {%- if conference.cfp.start -%},
```

File: upcoming.json
Comment (cketti) [inline upcoming.json:1]: Comments from `past.json` also apply here.

**Comment by ashdavies** [comment]
Thanks for the comments @cketti, I've applied your suggestions, the JSON looks much better now 👍

**Review by cketti — APPROVED** [review]
Nice :+1: 

Greetings from Ireland :shamrock:

**Pull request merged**